app/modules/asset_groups/resources.py

```diff
@@ -65,16 +64,13 @@ class AssetGroups(Resource):
             'action': AccessOperation.READ,
         },
     )
-    @api.parameters(PaginationParameters())
     @api.response(schemas.BaseAssetGroupSchema(many=True))
+    @api.paginate()
```

Comment:
So should everything now use this and not use @api.parameters(PaginationParameters()) 
I don't think I'm alone in not understanding the difference between these two.

Reply:
The `@api.paginate` decorator is very similar to `@api.parameters` but is unique in two different ways: 
1. It is optional to specify the Parameter class with @paginate because it assumes the parameters are `PaginationParameters()`.  With the `@parameters` decorator, you must specify which parameters Class should be used with a given API.
2. The `@paginate` decorator will automatically parse and perform the appropriate SQLAlchemy functions on a query to support pagination (offset, limit, sorting, reverse).  For this reason, the `@paginate` decorator expects that the function will return an SQLAlchemy query, not an object or list of objects.  

This is also why the `@api.paginate` decorator is now below the `@api.response` decorator, because the response schema will serialize the results produced by the pagination code.  If we had the pagination decorator above the response schema, it would serialize the output first and then we would perform paging after we have loaded the content out of the database.  This is bad because it would be slow (we want pagination to be done at the same time the SELECT query happened).

Reply:
We only want to use the `@api.paginate` decorator over the `@api.parameters` for listing and sorting APIs, where we would want to page the results.  All other APIs should continue to use `@api.parameters`.

Reply:
Understood. It might be worth updating the boilerplate templates for this too but to be fair, this PR is big enough already, that can probably be done under a different PR.

Reply:
I've updated the boilerplate templates to include the pagination changes.